ghost/core/core/server/services/members/jobs/clean-expired-comped.js

```diff
@@ -0,0 +1,71 @@
+const {parentPort} = require('worker_threads');
+const debug = require('@tryghost/debug')('jobs:clean-expired-comped');
+const moment = require('moment');
+
+// recurring job to clean expired complimentary subscriptions
+
+// Exit early when cancelled to prevent stalling shutdown. No cleanup needed when cancelling as everything is idempotent and will pick up
+// where it left off on next run
+function cancel() {
+    if (parentPort) {
+        parentPort.postMessage('Expired complimentary subscriptions cleanup cancelled before completion');
+        parentPort.postMessage('cancelled');
+    } else {
+        setTimeout(() => {
+            process.exit(0);
+        }, 1000);
+    }
+}
+
+if (parentPort) {
+    parentPort.once('message', (message) => {
+        if (message === 'cancel') {
+            return cancel();
+        }
+    });
+}
+
+(async () => {
```

Comment:
While we are at it, @rishabhgrg can you please check if we still need this self-execution function syntax. AFAIR our current range of Node versions supports top level await :thinking:

Reply:
From a quick check, top-level await seems to be still [only available in ES modules so we'll probably have to keep this style for now, but I'll come back to it later today to make sure that's the case.